test/signalrclient-e2e-tests/hub_connection_tests.cpp

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include "stdafx.h"
+#include <string>
+#include "cpprest\details\basic_types.h"
+#include "cpprest\json.h" 
+#include "connection.h"
+#include "hub_connection.h"
+#include "url.h"
+
+TEST(hub_connection_tests, connection_status_start_stop_start_reconnect)
+{
+    auto hubConn = std::make_shared<signalr::hub_connection>(url + U("SignalR"));
+    auto weak_hubConn = std::weak_ptr<signalr::hub_connection>(hubConn);
+    auto reconnecting_event = std::make_shared<pplx::event>();
+    auto reconnected_event = std::make_shared<pplx::event>();
+
+    auto hubProxy = hubConn->create_hub_proxy(U("hubConnection"));
+
+    hubConn->set_reconnecting([weak_hubConn, reconnecting_event]()
+    {
+        ASSERT_EQ(weak_hubConn.lock()->get_connection_state(), signalr::connection_state::reconnecting);
+        reconnecting_event->set();
+    });
+
+    hubConn->set_reconnected([weak_hubConn, reconnected_event]()
+    {
+        ASSERT_EQ(weak_hubConn.lock()->get_connection_state(), signalr::connection_state::connected);
+        reconnected_event->set();
+    });
+
+    try
+    {
+        hubConn->start().get();
+        ASSERT_EQ(hubConn->get_connection_state(), signalr::connection_state::connected);
+
+        hubConn->stop().get();
+        ASSERT_EQ(hubConn->get_connection_state(), signalr::connection_state::disconnected);
+
+        hubConn->start().get();
+        ASSERT_EQ(hubConn->get_connection_state(), signalr::connection_state::connected);
+    }
+    catch (...)
+    {
+        ASSERT_TRUE(false);
+    }
+
+    try
+    {
+        web::json::value obj{};
+        obj[0] = web::json::value(U("test"));
+
+        hubProxy.invoke<web::json::value>(U("forceReconnect")).get();
+    }
+    catch (...)
+    {
+    }
+
+    ASSERT_FALSE(reconnecting_event->wait(40000));
```

Comment:
40 seconds seems very long... because we run these tests locally shouldn't we expect this to finish much faster? (same below)